src/Compilers/Core/Portable/Text/TextSpan.cs

```diff
@@ -35,6 +35,26 @@ public TextSpan(int start, int length)
             Length = length;
         }
 
+        /// <summary>
+        /// Internally, the TextChangeRange merging algorithm must occasionally create TextSpans which are relative to
+        /// some delta such that their own start position is negative, but their final adjusted start position is non-negative.
+        /// </summary>
```

Comment:
i *hate* this.  IMO, don't use TextSpan for this.  use anotehr type.  the reason for this is that TextSpan *till now* has strongly enforced an invariant that is now baked into everyone's mental model.  if you break that, it makes TextSpan much harder to understand in the future (including in your PR).  Indeed, till now, i though TextSpan's invariants all held.

If you really want to have a start/end that doesn't ahve these invaraitns then define a new type, or just use a tuple.  

THis is a dig-my-heels in point :)

Reply:
Have introduced a different type to use instead in c8fd0555d65dee43abbd50d0570020e5e4e2f7c2. Thanks